index.js

```diff
@@ -60,31 +60,51 @@ class tracetoWeb3{
     return '0x'+eutil.privateToPublic(this.web3.eth.accounts.wallet[0].privateKey).toString('hex');
   }
 
-  callContractbyIdx(idx, funcName, callback, ...paras){
-    return this.contracts[idx].methods[funcName](...paras).call({}, callback);
+  callContractbyIdx(idx, funcName, ...paras){
+    const _this = this;
+    return new Promise((resolve, reject) => {
+      _this.contracts[idx].methods[funcName](...paras).call({}, (err, data) => {
+        if(!err)
+          resolve(data);
+        else
+          reject(err);
```

Comment:
Are we handling this error elsewhere?

Reply:
it will be catch while BE using the workflow